docs/source/conf.py

```diff
@@ -30,8 +30,18 @@
     'sphinxcontrib.bibtex',
     'sphinx.ext.napoleon',
     'sphinx.ext.viewcode',
+    'sphinx.ext.duration',
+    'sphinx.ext.doctest',
+    'sphinx.ext.inheritance_diagram',
+    'sphinx.ext.autosectionlabel',
```

Comment:
Are all of these new extensions still needed? They were added in the commits before using the autodoc capability, so I feel compelled to ask.